pkg/adaptor/mongodb/mongodb.go

```diff
@@ -397,14 +398,56 @@ func (m *MongoDB) catData() error {
 	return nil
 }
 
+func (m *MongoDB) catQuery(collection string, lastID interface{}, sess *mgo.Session) *mgo.Query {
+	query := bson.M{}
+	if lastID != nil {
+		query = bson.M{"_id": bson.M{"$gte": lastID}}
+	}
+	return sess.DB(m.database).C(collection).Find(query).Sort("_id")
+}
+
+func (m *MongoDB) requeryable(collection string) bool {
+	sess := m.mongoSession.Copy()
+	defer sess.Close()
+	indexes, err := sess.DB(m.database).C(collection).Indexes()
+	if err != nil {
+		fmt.Printf("[ERROR] unable to list indexes for %s, %s", collection, err)
+		return false
+	}
+	for _, index := range indexes {
+		if index.Key[0] == "_id" {
+			var result bson.M
+			err := sess.DB(m.database).C(collection).Find(nil).Select(bson.M{"_id": 1}).One(&result)
+			if err != nil {
+				fmt.Printf("[ERROR] unable to sample document, %s", err)
+				break
+			}
+			if id, ok := result["_id"]; ok && sortable(id) {
+				return true
+			}
+			break
+		}
+	}
+	fmt.Printf("[WARN] %s invalid _id, any issues with copying the collection will be aborted", collection)
+	return false
+}
+
+func sortable(id interface{}) bool {
+	switch id.(type) {
+	case bson.ObjectId, string, float64, time.Time:
```

Comment:
from the testing I did, int's seem to always come back as float64 but I'll throw `int64` in there just to be safe